std/src/main/scala/algebra/std/map.scala

```diff
@@ -130,3 +133,25 @@ class MapRng[K, V](implicit V: Rng[V]) extends MapSemiring[K, V] with Rng[Map[K,
   override def minus(x: Map[K, V], y: Map[K, V]): Map[K, V] =
     subtractMap(x, y)(V.minus)(V.negate)
 }
+
+class MapGenBool[K, V](implicit GBV: GenBool[V], EqV: Eq[V]) extends GenBool[Map[K, V]] {
+  def zero: Map[K, V] = Map.empty
+  
+  def and(a: Map[K, V], b: Map[K, V]): Map[K, V] = (
+    (a.keySet intersect b.keySet)
+    map { k => (k, GBV.and(a(k), b(k))) }
+    filter { (kv: (K, V)) => EqV.neqv(GBV.zero, kv._2) }
+  ).toMap
+
+  def or(a: Map[K, V], b: Map[K, V]): Map[K, V] = (
+    (a.keySet union b.keySet)
+    map { k => (k, GBV.or(a.getOrElse(k, GBV.zero), b.getOrElse(k, GBV.zero))) }
+  ).toMap
+
+  def without(a: Map[K, V], b: Map[K, V]): Map[K, V] = (
+    a.iterator map { kv => b.get(kv._1) match {
```

Comment:
I think the absence of the `.` actually changes the order in which scala infers types. Except for operators, I always prefer the `.` for this reason.